### PR TITLE
Yesterdays games

### PR DESCRIPTION
**Pull request by ggoolsby**
Pulls from baseball reference and creates data frame with games, including team name, score, and pitcher of record.

screen shot 2018-08-05 at 5 36 44 pm

### CODE REVIEW

File: yesterdaysGames.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+"""
+Gray Goolsby
+8/5/2018
+
+Pulls game results from previous day
```
Comment (ham-evans) [inline yesterdaysGames.py]: 'Pulls all MLB results from previous day' maybe?

File: yesterdaysGames.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+def pullGameTable(url):
+    soup = pullSoup(url)
+    
+    allTables = soup.findAll('div', {"class": "game_summaries"})
+    gameTables = allTables[0].findAll('table')
+    games = []
+
+    for i in range(len(gameTables)):
+        gameStats = gameTables[i].findAll('td')
```
Comment (ham-evans) [inline yesterdaysGames.py]:
```
for element in gameTables: 
    gameStats = element.findAll('td')
```
You could do something like that instead- a little cleaner

File: yesterdaysGames.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        
+        gameStats = str(gameStats)
+        gameStats = gameStats.split(',')
```
Comment (ham-evans) [inline yesterdaysGames.py]: Combine into 1 line

File: yesterdaysGames.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        
+        if(i%2 == 0):
+            t1 = gameStats[0]
+            t1S = gameStats[1]
+            t2 = gameStats[3]
+            t2S = gameStats[4]
```
Comment (ham-evans) [inline yesterdaysGames.py]: Maybe name those something that is more clear ^^^^

File: yesterdaysGames.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            startTeam = t1.find('>')
+            startScore = t1S.find('>')
```
Comment (ham-evans) [inline yesterdaysGames.py]: can combine into 1 line above^
`firstTeam = gameStats[0].find('>')`

File: yesterdaysGames.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            
+            t1 = t1[startTeam+33:len(t1)-9]
+            t2 = t2[startTeam+33:len(t2)-9]
```
Comment (ham-evans) [inline yesterdaysGames.py]: change- need to generalize more

File: yesterdaysGames.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            
+            t1S = t1S[startScore+1:len(t1S)-5]
+            t2S = t2S[startScore+1:len(t2S)-5]
```
Comment (ham-evans) [inline yesterdaysGames.py]: same here ^^

File: yesterdaysGames.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            wPitcher = gameStats[1]
+            lPitcher = gameStats[3]
+            wPitcher = wPitcher[5:wPitcher.rfind(')')+1]
+            lPitcher = lPitcher[5:lPitcher.rfind(')')+1]
+            if(t1S>t2S):
+                wTeam = [t1, t1S, wPitcher]
+                lTeam = [t2, t2S, lPitcher]
+            else:
+                wTeam = [t2,t2S,wPitcher]
+                lTeam = [t1,t1S,lPitcher]
```
Comment (ham-evans) [inline yesterdaysGames.py]: take this out of else statement

File: yesterdaysGames.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            t2S = t2S[startScore+1:len(t2S)-5]
+            
+            t1S = int(t1S)
+            t2S = int(t2S)
+            
+        else:
+            wPitcher = gameStats[1]
+            lPitcher = gameStats[3]
+            wPitcher = wPitcher[5:wPitcher.rfind(')')+1]
+            lPitcher = lPitcher[5:lPitcher.rfind(')')+1]
```
Comment (ham-evans) [inline yesterdaysGames.py]: try to combine into 1 line- harder than above

File: yesterdaysGames.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
+"""
+Gray Goolsby
+8/5/2018
+
+Pulls game results from previous day
+"""
+import pandas as pd
+from pullWebData import pullSoup
+
+def pullGameTable(url):
```
Comment (ham-evans) [inline yesterdaysGames.py]: Maybe consolidate functions- call this `yesterdayGames ():`

no need to give it url if always using same one- just define url on first line of function

File: yesterdaysGames.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+            else:
+                wTeam = [t2,t2S,wPitcher]
+                lTeam = [t1,t1S,lPitcher]
+            game = [wTeam,lTeam]
+            games.append(game)
+    games = pd.DataFrame(games)
+    games.columns = ['W', 'L']
+    return games
+
+def yesterdaysGames():
```
Comment (ham-evans) [inline yesterdaysGames.py]: function not needed

File: yesterdaysGames.py (hunk 2 above)
Comment (ggoolsby) [reply yesterdaysGames.py]: can't because I use i as tracker to distinguish between games and pitchers

File: yesterdaysGames.py [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+            team2Score = int(gameStats[4][gameStats[4].find('>')+1:gameStats[4].rfind('<')])
+            
+            if(team1Score>team2Score):
+                winningTeam = [team1, team1Score]
+                losingTeam = [team2, team2Score]
+            else:
+                winningTeam = [team2, team2Score]
+                losingTeam = [team1, team1Score]
+
+        else:
```
Comment (ham-evans) [inline yesterdaysGames.py:35]: is this else statement necessary

Comment (ggoolsby) [reply yesterdaysGames.py:35]: yes

**Review by ham-evans — APPROVED** [review]

**Pull request merged**